gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -139,26 +139,51 @@ def _get_http_options():
     return response
 {% endmacro %}
 
-{% macro rest_call_method_common(body_spec, method_name, service_name) %}
+{# rest_call_method_common includes the common code for a rest __call__ method to be
+    re-used for sync REST, async REST, and mixins __call__ implementation.
+
+    Args:
+        body_spec (str): The http options body i.e. method.http_options[0].body
+        method_name (str): The method name.
+        service_name (str): The service name.
+        is_async (bool): Used to determine the code path i.e. whether for sync or async call. #}
+{% macro rest_call_method_common(body_spec, method_name, service_name, is_async=False) %}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_class_prefix = "Async" if is_async else "" %}
 
     http_options = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_http_options()
+    {% if not is_async %}
+    {# TODO (ohmayr): Make this unconditional once REST interceptors are supported for async. See issue: b/362949568 #}
```

Comment:
We really shouldn't put links to internal issues in open-source code, but since this is meant to be temporary, it's alright...though I would annotate a bit further:

```suggestion
    {# TODO (ohmayr): Make this unconditional once REST interceptors are supported for async. Googlers, see internal tracking issue: b/362949568 #}
```

Reply:
addressed.